lib/migrate.js

```diff
@@ -46,11 +46,10 @@ function migrate (set, direction, migrationName, fn) {
     if (isPromise) {
       returnValue
         .then(completeMigration)
-        .catch(completeMigration)
+        .catch(fn)
```

Comment:
This haven't changed, but just realised that this will swallow any exceptions thrown in the `completeMigration` or `fn` functions, which will be quite unexpected if used like this:

```
migrate(..., (err) => {
  if (err) throw err

  console.log('All done')
})
```

In this case, the `throw err` will not terminate Node.js, but rather just cause the promise created here to reject

Reply:
Yeah I think you are correct.  Sorry I dont use `promises` so I didn't create test cases for these when I was pulling it all together.  Would love for someone who does use promises to take a TLC pass on this.

Reply:
To be honest, I would prefer just switching to using promises everywhere. The shim for the migrations could then be:

```js
new Promise((resolve, reject) => {
  const returnValue = migration[direction]((err) => {
    if (err) return reject(err)

    resolve()
  })

  // Is it a promise?
  isPromise = typeof Promise !== 'undefined' && returnValue instanceof Promise
 
  // If not a promise and arity is not 1, warn
  if (!isPromise && arity < 1) set.emit('warning', 'it looks like your migration did not take or callback or return a Promise, this might be an error')
 
  // Settle with the provided promise
  if (isPromise) resolve(returnValue)
})
```

Reply:
Yeah, in this case it seems like it would be much more simple to wrap them it in a promise than to unwrap it from the promise.  This seems like a good change.